src/Filesystem/Filesystem.php

```diff
@@ -231,6 +231,34 @@ public function isLocalDisk(\Illuminate\Filesystem\FilesystemAdapter $disk): boo
         return ($disk->getAdapter() instanceof \League\Flysystem\Local\LocalFilesystemAdapter);
     }
 
+    /**
+     * Apply a unique index to a filename from provided list i.e.
+     * winter.txt, [winter_1.txt, winter_2.txt] -> winter_3.txt
+     * winter.txt, [winter_1.txt, winter_3.txt] -> winter_4.txt
```

Comment:
```suggestion
     * winter.txt, [winter.txt] -> winter_1.txt
     * winter.txt, [winter.txt, winter_1.txt, winter_2.txt] -> winter_3.txt
     * winter.txt, [winter.txt, winter_1.txt, winter_3.txt] -> winter_4.txt
     * winter.txt, [winter_1.txt, winter_3.txt] -> winter.txt
```

@damsfx it should probably allow the input to be returned unmodified if it doesn't exist within the array.

Reply:
@LukeTowers If the entry doesn't exist in the array, it must be returned with another index anyway, right?

Reply:
@damsfx I'm making the suggestion that if it doesn't exist in the array it should be returned unmodified. The method is intended to give you a unique filename provided the input of a desired filename and a list of already existing options. If the filename provided is already unique, then there's no need to modify it.

Reply:
@LukeTowers So in a such case, the same filename is needed in the array of references to get the next incremented name.

Reply:
The suggested change above should list all the cases that I think the method should need to handle, I've already updated the tests so it should just require a minor change in the method itself.

Reply:
@LukeTowers Your modifications in tests goes to a failure for me.

```php
// File already unique, return original
'winter.cms' => ['winter.cms', ['winter_1.cms']],
```
**FAILED !**
The function return `wintert_2.cms` in that case.  

What I don't understand is why you want to return the unmodified value if it's unique.
In any case, we should return the incremented value! 🤯
```php
(new Filesystem())->unique('winter.cms', ['test.cms']);   // winter_1.cms
(new Filesystem())->unique('winter.cms', []);             // winter_1.cms
```

Reply:
@damsfx The tests are failing because the method doesn't currently have the logic to allow the name to be returned unmodified. The method claims to return a unique filename given the input of a desired filename and a list of existing filenames; why does it have to always return an incremented value if the desired filename isn't present?

Reply:
@LukeTowers `Apply a unique index to a filename from provided list ...`  
Maybe we should then say `Apply a unique index to a filename from provided list ONLY if it is found in the list ...` or change the name of the function which is perhaps the source of the confusion.

In any case, the current function respects the logic of the original one: 
https://github.com/wintercms/storm/commit/68bb7acc6c0a1377b375d3dc9052310a9d8d02d8

Reply:
@LukeTowers Do we continue with this PR in draft?